pkg/deploy/spindeploy/changedetector/config.go

```diff
@@ -25,7 +27,15 @@ func (ch *configChangeDetector) IsSpinnakerUpToDate(ctx context.Context, spinSvc
 	if err != nil {
 		return false, err
 	}
-	eh := spinSvc.GetStatus().LastConfigHash
-	spinSvc.GetStatus().LastConfigHash = h
+	st := spinSvc.GetStatus()
```

Comment:
I see that this function is doing a modification of the status as well as checking if there were any changes.
What happens if the config was changed, this function sets the new hash, but then there's an error applying the changes, would the lastDeployedHashes be out of sync with what is actually deployed? Would it be better to move updating the last deployed hash after the deployment is applied?

Reply:
Then the status wouldn't be saved. We're working on a clone here (normally).

Reply:
To clarify the status should be updated as part of `commitConfigToStatus` after all manifests are saved.